lib/azuremetadata/azuremetadata.py

```diff
@@ -133,3 +139,23 @@ def _make_request(url):
         print("An error occurred when fetching metadata:", file=sys.stderr)
         print(last_error, file=sys.stderr)
         return {}
+
+    @staticmethod
+    def _get_api(api_version):
+        if api_version == 'latest':
+            api_newest_versions = AzureMetadata._get_api_newest_versions()
+            api_version = api_newest_versions[0]
```

Comment:
The method is called `_get_api_newest_versions`, but it actually returns the list of all available versions.
Rename to `_get_api_newest_version` and actually return one newest version, instead of a list?

Reply:
> The method is called `_get_api_newest_versions`, but it actually returns the list of all available versions.

If I understood the response correctly, the [query](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/instance-metadata-service#versioning) without api version returns the _newest_ versions, meaning there are other older versions available that can be used in the query, not sure if there is a way to retrieve those or if treating that response as _all_ available versions, then.
 
> Rename to `_get_api_newest_version` and actually return one newest version, instead of a list?

I think it should return a list (thinking of #11), as I mentioned, I'm not sure if there is another way to get _all_ available API versions or the newest (with that query)